### PR TITLE
Cleanup workflow - remove PSProxyJob

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Remove PSProxyJob class.

## PR Context

Related #9570.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/System.Management.Automation/engine/hostifaces/PowerShell.cs [hunk 1]
```diff
@@ -3896,29 +3896,6 @@ private void ResetOutputBufferAsNeeded()
             }
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
TabCompletion.Miscellaneous completion tests.Test member completion of a static method invocation
```powershell
Expected a collection with size 34, but got collection with size 33 @(System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult).
at <ScriptBlock>, D:\a\1\s\test\powershell\Host\TabCompletion\TabCompletion.Tests.ps1: line 757
757:             $res.CompletionMatches | Should -HaveCount 34
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
TabCompletion.Miscellaneous completion tests.Test member completion of a static method invocation
```powershell
Expected a collection with size 34, but got collection with size 33 @(System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult).
at <ScriptBlock>, /Users/vsts/agent/2.153.2/work/1/s/test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1: line 757
757:             $res.CompletionMatches | Should -HaveCount 34
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/hostifaces/PowerShell.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
TabCompletion.Miscellaneous completion tests.Test member completion of a static method invocation
```powershell
Expected a collection with size 34, but got collection with size 33 @(System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult, System.Management.Automation.CompletionResult).
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1: line 757
757:             $res.CompletionMatches | Should -HaveCount 34
```

**Review by PaulHigin — APPROVED** [review]

**Comment by iSazonov** [comment]
@SteveL-MSFT @adityapatwardhan Removing PSProxyJob class is formally a breaking change and it should be removed from SDK too.
Do we need PowerShell Committee approval?

**Comment by SteveL-MSFT** [comment]
@iSazonov the committee has already agreed to remove the workflow code where many of it is part of the public API.  I don't believe this is any different.

**Pull request merged**